src/oas/__tests__/operation.test.ts

```diff
@@ -162,4 +162,85 @@ describe('oas operation', () => {
       },
     });
   });
+  it('uses existing stableId', () => {
+    const document = {
+      openapi: '3.1.0',
+      'x-stoplight': {
+        id: 'tohfdtsy98xjv',
+      },
+      info: {
+        title: '1st',
+        version: '1.0',
+      },
+      paths: {
+        '/test': {
+          get: {
+            summary: 'Your GET endpoint',
+            tags: [],
+            responses: {},
+            operationId: 'get-test',
+            requestBody: {
+              content: {
+                'application/json': {
+                  schema: {
+                    type: 'object',
+                    properties: {},
+                  },
+                },
+              },
+            },
+            'x-stoplight': {
+              id: 't44kj45klj46v',
+            },
+          },
+        },
+      },
+    };
+    const result = transformOas3Operations(document);
+
+    expect(result).toStrictEqual([
+      {
+        id: 't44kj45klj46v',
+        iid: 'get-test',
+        method: 'get',
+        path: '/test',
+        request: {
+          body: {
+            id: 'http_request_body-t44kj45klj46v',
+            contents: [
+              {
+                id: 'http_media-http_request_body-t44kj45klj46v-application/json',
+                mediaType: 'application/json',
+                encodings: [],
+                examples: [],
+                schema: {
+                  $schema: 'http://json-schema.org/draft-07/schema#',
+                  type: 'object',
+                  properties: {},
+                  'x-stoplight': {
+                    id: 'schema-http_media-http_request_body-t44kj45klj46v-application/json-',
```

Comment:
the trailing dash on this still is alluding me - its not there in the join for generating the id
<img width="1476" alt="Screenshot 2023-07-07 at 8 59 45 AM" src="https://github.com/stoplightio/http-spec/assets/19455540/836ba991-3581-4f6d-9187-43ae2307a989">

Reply:
gotcha! thanks!